src/features/events/components/EventActionButtons.tsx

```diff
@@ -43,7 +43,9 @@ const EventActionButtons: React.FunctionComponent<EventActionButtonsProps> = ({
     setPublished(null);
   };
   const handleChangeDate = (date: string | null) => {
-    setPublished(date);
+    if (date) {
```

Comment:
I think this should be compared with the previous value. Because right now, I can still reproduce the bug if I publish the event first. That way, opening the date picker and clicking outside of it calls this function with the date that was already there, and that reproduces the bug.

Reply:
I have added a check against the previous date as you suggested.

However I have now noticed that if you set a date further ahead in the future it changes the date to scheduled and that 'Scheduled' remains as the event status regardless if you cancel or restore. 

Should future cancelled events show as scheduled? It seems a bit unintuitive..

Reply:
Nice work trying different edge cases! It sounds to me like `getEventState()` needs to be improved. Ideally, there should be tests for it that test these cases. Do you want to check it out?

Reply:
Yes, I'll have a look! :)